src/components/MultiHopTrade/hooks/useQuoteValidationPredicateObject.tsx

```diff
@@ -0,0 +1,34 @@
+import { useHopHelper } from 'components/MultiHopTrade/hooks/useHopHelper'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+
+type QuoteValidationPredicateObject = Record<string, boolean>
+
+export const useQuoteValidationPredicateObject = (): QuoteValidationPredicateObject => {
+  const {
+    sellAssetBalanceCryptoBaseUnit,
+    sellAmountCryptoBaseUnit,
+    firstHopFeeAssetBalancePrecision,
+    firstHopNetworkFeeCryptoPrecision,
+    firstHopTradeDeductionCryptoPrecision,
+    lastHopFeeAssetBalancePrecision,
+    lastHopNetworkFeeCryptoPrecision,
+  } = useHopHelper()
+
+  const hasSufficientSellAssetBalance = bnOrZero(sellAssetBalanceCryptoBaseUnit).gte(
+    bnOrZero(sellAmountCryptoBaseUnit),
+  )
+  const firstHopHasSufficientBalanceForGas = bnOrZero(firstHopFeeAssetBalancePrecision)
+    .minus(firstHopNetworkFeeCryptoPrecision)
+    .minus(firstHopTradeDeductionCryptoPrecision)
+    .gte(0)
+
+  const lastHopHasSufficientBalanceForGas = bnOrZero(lastHopFeeAssetBalancePrecision)
+    .minus(lastHopNetworkFeeCryptoPrecision)
+    .gte(0)
```

Comment:
these ones are slightly more sensible to put in a useMemo/selector

Reply:
I'm going to pragmatically leave these as-is since we're going to move it all to a slice anyway.